gobblin-api/src/main/java/gobblin/configuration/SourceState.java

```diff
@@ -67,13 +68,17 @@ public SourceState(State properties, List<WorkUnitState> previousTaskStates) {
     this.previousTaskStates.addAll(previousTaskStates);
   }
 
+  public void setWatermarkInterval(WatermarkInterval interval) {
```

Comment:
You should be able to set the `WatermarkInterval` using the `WorkUnit(SourceState state, Extract extract, WatermarkInterval watermarkInterval)` constructor. I think logically it makes more sense to specify the `WatermarkInterval` on a `WorkUnit` level than a `SourceState` level because a `WorkUnit` a single `WorkUnit` is responsible for pulling data from a low `Watermark` to a high `Watermark`.